hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/realtime/RealtimeCompactedRecordReader.java

```diff
@@ -198,7 +197,7 @@ private Option<HoodieAvroIndexedRecord> mergeRecord(HoodieRecord<?> newRecord, A
     GenericRecord genericRecord = HiveAvroSerializer.rewriteRecordIgnoreResultCheck(oldRecord, getLogScannerReaderSchema());
     HoodieRecord record = new HoodieAvroIndexedRecord(genericRecord);
     Option<Pair<HoodieRecord, Schema>> mergeResult = HoodieAvroRecordMerger.INSTANCE.merge(record,
-        genericRecord.getSchema(), newRecord, getLogScannerReaderSchema(), new TypedProperties(payloadProps));
+        genericRecord.getSchema(), newRecord, getLogScannerReaderSchema(), payloadProps);
```

Comment:
any reason why we were wrapping in TypedProperties before? Does it need to be serializable?

Reply:
`TypedProperties` itself is serializable, it should be a mistake, in HUDI-6937, we found a performance regression of almost 100% for parquet file from insert handle.